distiller/scheduler.py

```diff
@@ -106,12 +106,12 @@ def add_policy(self, policy, epochs=None, starting_epoch=0, ending_epoch=1, freq
                                        'ending_epoch': ending_epoch,
                                        'frequency': frequency}
 
-    def on_epoch_begin(self, epoch, optimizer=None):
-        if epoch in self.policies:
-            for policy in self.policies[epoch]:
-                meta = self.sched_metadata[policy]
-                meta['current_epoch'] = epoch
-                policy.on_epoch_begin(self.model, self.zeros_mask_dict, meta)
+    def on_epoch_begin(self, epoch, optimizer=None, **kwargs):
+        for policy in self.policies.get(epoch, list()):
+            meta = self.sched_metadata[policy]
+            meta['current_epoch'] = epoch
+            policy.on_epoch_begin(self.model, self.zeros_mask_dict, meta,
+                                  epoch=epoch, **kwargs)
```

Comment:
Epoch number is in `meta`, no need to pass it separately.
```suggestion
                                  **kwargs)
```